triton-vm/src/stark.rs

```diff
@@ -220,108 +253,171 @@ impl Stark {
+        prof_stop!(maybe_profiler, "LDE");
+        let fri_quotient_codeword_digests = fri_quotient_codeword
+            .iter()
+            .map(|&x| x.into())
+            .collect_vec();
+        let quot_merkle_tree: MerkleTree<StarkHasher, _> =
+            Maker::from_digests(&fri_quotient_codeword_digests);
+        let quot_merkle_tree_root = quot_merkle_tree.get_root();
+        proof_stream.enqueue(&ProofItem::MerkleRoot(quot_merkle_tree_root));
+        prof_stop!(maybe_profiler, "commit to quotient codeword");
+        debug_assert_eq!(self.fri.domain.length, quot_merkle_tree.get_leaf_count());
+
+        // Get weights for remainder of the combination codeword.
         prof_start!(maybe_profiler, "Fiat-Shamir");
-        let num_non_lin_combi_weights =
-            2 * (NUM_BASE_COLUMNS + NUM_EXT_COLUMNS + num_all_table_quotients());
-        let non_lin_combi_weights = proof_stream.sample_scalars(num_non_lin_combi_weights);
+        let num_base_and_ext_codeword_weights = NUM_BASE_COLUMNS + NUM_EXT_COLUMNS;
+        let base_and_ext_codeword_weights =
+            proof_stream.sample_scalars(num_base_and_ext_codeword_weights);
         prof_stop!(maybe_profiler, "Fiat-Shamir");
 
-        prof_start!(maybe_profiler, "nonlinear combination");
-        prof_start!(maybe_profiler, "create combination codeword");
-        let combination_codeword = self.create_combination_codeword(
+        prof_start!(maybe_profiler, "base&ext codeword: linear combination");
+        let base_and_ext_codeword = self.create_base_and_ext_codeword(
             quotient_domain,
             base_quotient_domain_codewords,
             extension_quotient_domain_codewords.slice(s![.., ..NUM_EXT_COLUMNS]),
-            master_quotient_table.view(),
-            &non_lin_combi_weights,
-            quotient_degree_bounds,
+            &base_and_ext_codeword_weights,
         );
-        prof_stop!(maybe_profiler, "create combination codeword");
+        prof_stop!(maybe_profiler, "base&ext codeword: linear combination");
 
-        prof_start!(maybe_profiler, "LDE 3");
-        let fri_combination_codeword_without_randomizer = Array1::from(
-            quotient_domain.low_degree_extension(&combination_codeword, self.fri.domain),
+        prof_start!(maybe_profiler, "DEEP: base&ext");
+        let base_and_ext_interpolation_poly = quotient_domain.interpolate(&base_and_ext_codeword);
+        let out_of_domain_next_row_base_and_ext_element =
+            base_and_ext_interpolation_poly.evaluate(&out_of_domain_value_next_row);
+        proof_stream.enqueue(&ProofItem::OutOfDomainElement(
+            out_of_domain_next_row_base_and_ext_element,
+        ));
+        let base_and_ext_next_row_deep_quotient = Self::deep_quotient(
+            &base_and_ext_codeword,
+            quotient_domain,
+            out_of_domain_value_next_row,
+            out_of_domain_next_row_base_and_ext_element,
         );
```

Comment:
I would prefer to avoid overloading the term "quotient" because then there will be quotient quotients.